src/main/scala/cromwell/webservice/PerRequest.scala

```diff
@@ -103,6 +103,14 @@ object PerRequest {
 trait PerRequestCreator {
   implicit def actorRefFactory: ActorRefFactory
 
-  def perRequest(r: RequestContext, props: Props, message: AnyRef, timeout: Duration = 1 minutes) =
-    actorRefFactory.actorOf(Props(new WithProps(r, props, message, timeout)))
+  def perRequest(r: RequestContext,
+                 props: Props, message: AnyRef,
+                 timeout: Duration = 1 minutes,
+                 name: String = PerRequestCreator.endpointActorName) = {
+    actorRefFactory.actorOf(Props(new WithProps(r, props, message, timeout, name)), name)
+  }
+}
+
+object PerRequestCreator {
+  def endpointActorName = "Endpoint-" + java.lang.Thread.currentThread.getStackTrace()(1).getMethodName + System.nanoTime()
```

Comment:
Heh, the previous convo with @mcovarr got dropped. I stole it from Agora and figure that unless we have a better idea we should probably just do what they're doing. At some point I'm going to look at putting in Monsanto's enhanced spray stuff which I think makes this simpler, I'll make a comment